File: software/core/Infrastructure/backend/Classes/sensor.js
class Sensor {
	constructor(name, value = null) {
		this._name = name;
		this._value = value;
	}

	get name() {
		return this._name;
	}

	get value() {
		return this._value;
	}

	set name(newName) {
		this._name = newName;
	}

	set value(newValue) {
		this._value = newValue;
	}
}

// const sensor1 = new Sensor("mysensor", 2);

// console.log(sensor1.name);
// console.log(sensor1.value);

// sensor1.name = "updatedsensor";
// sensor1.value = 3;

// console.log(sensor1.name);
// console.log(sensor1.value);

module.exports = Sensor;

File: software/core/Infrastructure/c++/classes/sensor.cpp
#include <iostream>
 
using namespace std;

class Sensor {
    protected:

        string _name;
        int _value;


    public:

        Sensor( string name, int value): _name(name), _value(value) {};

        void setName(string newName){
            _name = newName;
        }

        void setValue( int newValue){
            _value = newValue;
        }

        string getName(){
            return _name;
        }

        int getValue(){
            return _value;
        }

        string toString(){
            return (
            "{'name':" + _name + ",'value':" + to_string(_value) + "}"
            );
        }
};


class SpeedSensor: public Sensor{

};

class TemperatureSensor: public Sensor{

};

class BatterySensor : public Sensor{

};

class BrakeSensor : public Sensor{

};

class PositionSensor : public Sensor{
    protected:
        int _x;
        int _y;
    
    public:
        PositionSensor(int x, int y): _x(x), _y(y) {};

        void setX(int x){
            _x = x;
        }

        void setY( int y){
            _y = y;
        }

        int getX(){
            return _x;
        }

        int getY(){
            return _y;
        }

        string toString(){
            return (
                "[" + to_string(_x) + "," + to_string(_y) + "]"
            );
        }
};